starknet_devnet/fee_token.py

```diff
@@ -58,24 +59,28 @@ def contract_class(self) -> DeprecatedCompiledClass:
     async def _mimic_constructor(self):
         starknet: Starknet = self.starknet_wrapper.starknet
         await starknet.state.state.set_storage_at(
-            FeeToken.ADDRESS,
-            get_selector_from_name("ERC20_name"),
-            str_to_felt(FeeToken.NAME),
+            storage_domain=StorageDomain.ON_CHAIN,
```

Comment:
StorageDomain.ON_CHAIN means data on L1? Shouldn't we use StorageDomain.OFF_CHAIN? Shouldn't we also mention that `StorageDomain.ON_CHAIN` in the documentation?

Is that even important? Maybe not, I'm just asking.

Reply:
In cairo-lang, there is an internal assertion on the used storage domain - they assert that it's on_chain.

https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/business_logic/state/storage_domain.py